lisa/transformers/kernel_source_installer.py

```diff
@@ -171,17 +212,27 @@ def _modify_code(self, node: Node, code_path: PurePath) -> None:
             self._log.debug(f"modifying code by {modifier.type_name()}")
             modifier.modify()
 
-    def _build_code(self, node: Node, code_path: PurePath) -> None:
+    def _build_code(self, node: Node, code_path: PurePath, kconfig_path: str) -> None:
         self._log.info("building code...")
 
         uname = node.tools[Uname]
         kernel_information = uname.get_linux_information()
 
-        result = node.execute(
-            f"cp /boot/config-{kernel_information.kernel_version_raw} .config",
-            cwd=code_path,
-        )
-        result.assert_exit_code()
+        if kconfig_path and len(kconfig_path) > 0:
```

Comment:
It looks all logics are like `if` or `if/else` for dom0. So, it's better to implement a new Dom0Installer, which inherits from this installer.

Reply:
This is not specific to dom0. If we will be passing any config path from runbook , we should refer to is otherwise it will be default flow of referring config from /boot. We are making it generic so we have flexibility to pass config file dynamically from runbook. It can be any platform not just dom0